app/jobs/send_user_invitation_job.rb

```diff
@@ -0,0 +1,14 @@
+# frozen_string_literal: true
+
+class SendUserInvitationJob < ActiveJob::Base
+  queue_as :user_invites
+
+  def perform(invitation_id)
+    invitation = Invitation.find(invitation_id)
+
+    mailer_klass = account.provider? ? ProviderInvitationMailer : InvitationMailer
+    mailer_klass.invitation(invitation).deliver_now
+
+    invitation.update_column(sent_at: Time.zone.now)
+  end
```

Comment:
I have some comments about this:
1. What if the invitation does not exist anymore? Then you should rescue that exception and ignore it.
2. Why not enqueuing the whole object instead of the ID? then it will use GlobalID
3. This is the most important one... why not doing all this job inside the mailer directly? I do not see the benefit of adding an extra job to be a middle man

Reply:
I think the idea of using a job is to update the column async, after the email is sent, no?

Reply:
I do not understand the reason why not the Observer, since [it is documented even for rails 6](https://apidock.com/rails/v6.0.0/ActionMailer/Base) and does not seem deprecated at all.
I think that @hallelujah confused these observers, with [these observers](https://github.com/rails/rails-observers) that were removed from Rails already (but can work using that gem).

But if I am wrong about this, then IMO, you should just add the `invitation.update_column(sent_at: Time.zone.now)` inside the mailer itself, or another option is doing as u did here but doing `deliver_now!`with the `!`to raise an error if it fails.

Reply:
Right, sorry! I was confused, thanks @Martouta !